src/GatewayDest.js

```diff
@@ -8,7 +8,11 @@ export default class GatewayDest extends React.Component {
 
   static propTypes = {
     name: React.PropTypes.string.isRequired,
-    tagName: React.PropTypes.string
+    tagName: React.PropTypes.oneOfType([
+      React.PropTypes.string,
+      React.PropTypes.func,
+      React.PropTypes.instanceOf(React.Component)
```

Comment:
I think it only needs `React.PropTypes.func`: http://react.jsbin.com/jiluza/edit?js,console